server/src/main/java/org/opensearch/cluster/decommission/DecommissionService.java

```diff
@@ -440,8 +442,11 @@ private static void ensureToBeDecommissionedAttributeWeighedAway(ClusterState st
                 "no weights are specified to attribute [" + decommissionAttribute.attributeName() + "]"
             );
         }
+        // in case the weight is not set for the attribute value, then we know that attribute values was not part of discovered routing node
+        // attribute or forced awareness attribute and in that case, we are ok if the attribute's value weight is not set. But if it's set,
+        // its weight has to be zero
         Double attributeValueWeight = weightedRouting.weights().get(decommissionAttribute.attributeValue());
-        if (attributeValueWeight == null || attributeValueWeight.equals(0.0) == false) {
+        if (attributeValueWeight != null && attributeValueWeight.equals(0.0) == false) {
```

Comment:
Add an assert that if weight for an attribute value is not set the it is NOT a routing node. Can we add similar checks at all places where we are modifying weights and not weights corresponding to an attribute is set?

Reply:
Its tough to say which attributes weight must be set. Rather we can assert that all routing nodes weights are set. Now, weights gets updated only at central place in `WeightedRoutingService`, where I have added the method `ensureWeightsSetForAllDiscoveredAndForcedAwarenessValues` which does the same